blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServerBuilder.scala

```diff
@@ -91,7 +91,7 @@ import scodec.bits.ByteVector
   */
 class BlazeServerBuilder[F[_]] private (
     socketAddress: InetSocketAddress,
-    executionContext: ExecutionContext,
+    executionContextF: F[ExecutionContext],
```

Comment:
If we don't want this to be an effect, we could have an internal sum type: `FromAsync` and `ExplicitExecutionContext`.  We did something similar on an `SSLContext`.  I think this model is probably fine, but I've thought about making the builder values `val`s so people can introspect, print, etc, and a sum type would be cleaner there.

It's also a bincompat refactor to get there from here, so I think this model is fine.

Reply:
I like this very much. Thanks!